msgraph/role_eligibility_schedule_request.go

```diff
@@ -0,0 +1,120 @@
+		ConsistencyFailureFunc: RetryOn404ConsistencyFailureFunc,
+		OData:                  query,
+		ValidStatusCodes:       []int{http.StatusOK},
+		Uri: Uri{
+			Entity: fmt.Sprintf("/roleManagement/directory/roleEligibilityScheduleRequests/%s", id),
+		},
+	})
+	if err != nil {
+		return nil, status, fmt.Errorf("RoleEligibilityScheduleRequestClient.BaseClient.Get(): %v", err)
+	}
+
+	defer resp.Body.Close()
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, status, fmt.Errorf("io.ReadAll(): %v", err)
+	}
+
+	var dirRole UnifiedRoleEligibilityScheduleRequest
+	if err := json.Unmarshal(respBody, &dirRole); err != nil {
+		return nil, status, fmt.Errorf("json.Unmarshal(): %v", err)
+	}
+
+	return &dirRole, status, nil
+}
+
+// Create creates a new UnifiedRoleEligibilityScheduleRequest.
+func (c *RoleEligibilityScheduleRequestClient) Create(ctx context.Context, resr UnifiedRoleEligibilityScheduleRequest) (*UnifiedRoleEligibilityScheduleRequest, int, error) {
+	var status int
+
+	resr.Action = &actionAdminAssign
+	body, err := json.Marshal(resr)
+	if err != nil {
+		return nil, status, fmt.Errorf("json.Marshal(): %v", err)
+	}
+
+	resp, status, _, err := c.BaseClient.Post(ctx, PostHttpRequestInput{
+		ConsistencyFailureFunc: RetryOn404ConsistencyFailureFunc,
+		Body:                   body,
+		ValidStatusCodes:       []int{http.StatusCreated},
+		Uri: Uri{
+			Entity: "/roleManagement/directory/roleEligibilityScheduleRequests",
+		},
+	})
+	if err != nil {
+		return nil, status, fmt.Errorf("RoleEligibilityScheduleRequestClient.BaseClient.Post(): %v", err)
+	}
+
+	defer resp.Body.Close()
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, status, fmt.Errorf("io.ReadAll(): %v", err)
+	}
+
+	var newRoleAssignment UnifiedRoleEligibilityScheduleRequest
+	if err := json.Unmarshal(respBody, &newRoleAssignment); err != nil {
+		return nil, status, fmt.Errorf("json.Unmarshal(): %v", err)
+	}
+
+	return &newRoleAssignment, status, nil
+}
+
+// Delete removes a UnifiedRoleEligibilityScheduleRequest.
+func (c *RoleEligibilityScheduleRequestClient) Delete(ctx context.Context, id string) (int, error) {
+	resr, status, err := c.Get(ctx, id, odata.Query{})
+	if err != nil {
+		return status, fmt.Errorf("RoleEligibilityScheduleRequestClient.Get(): %v", err)
+	}
+	resrBody := UnifiedRoleEligibilityScheduleRequest{
+		Action:           &actionAdminRemove,
+		RoleDefinitionId: resr.RoleDefinitionId,
+		DirectoryScopeId: resr.DirectoryScopeId,
+		PrincipalId:      resr.PrincipalId,
+	}
+	body, err := json.Marshal(resrBody)
+	if err != nil {
+		return status, fmt.Errorf("json.Marshal(): %v", err)
+	}
+	_, status, _, err = c.BaseClient.Post(ctx, PostHttpRequestInput{
+		Body:             body,
+		ValidStatusCodes: []int{http.StatusCreated},
+		Uri: Uri{
+			Entity: "/roleManagement/directory/roleEligibilityScheduleRequests",
+		},
+	})
+	if err != nil {
+		return status, fmt.Errorf("RoleAssignments.BaseClient.Get(): %v", err)
```

Comment:
We'll want to check through all the error messages to make sure they reflect the correct client and method.